fastify.js

```diff
@@ -482,19 +480,23 @@ function build (options) {
   }
 
   function inject (opts, cb) {
-    if (!shot) throw new Error('"shot" library is not installed: "npm install shot@3 --save-dev"')
-
-    const waitingForReadyEvent = started
-      ? Promise.resolve()
-      : new Promise((resolve, reject) => this.ready(err => (err ? reject : resolve)(err)))
-    const injectPromise = waitingForReadyEvent
-      .then(() => new Promise(resolve => shot.inject(this, opts, resolve)))
+    if (started) {
+      return Inject(this, opts, cb)
+    }
 
     if (cb) {
-      injectPromise.then(cb, cb)
-      return
+      this.ready(err => {
+        if (err) throw err
+        return Inject(this, opts, cb)
+      })
+    } else {
+      return new Promise((resolve, reject) => {
+        this.ready(err => {
+          if (err) return reject(err)
+          resolve(Inject(this, opts))
```

Comment:
I don't like this.

More better
```
return new Promise((resolve, reject) => {
  this.ready(err => {
    if (err) return reject(err)
    resolve()
})
  .then(() => Inject(this, opts))
```

Reply:
I don't know, is just one line more to get the same functionality.
Why do you prefer add another `then`?

Reply:
Just only for separation.
The first promise resolves the "ready" waiting. The second one fires the injection.

Reply:
I think we should always return a promise